### PR TITLE
Disable implicit namespace imports

### PR DESCRIPTION
**Pull request by safern**
In preview7 SDK some namespace imports are defined by default. Most of our projects don't use implicit references as we are low in the stack and producing those references. This can cause issues when building with a preview7 SDK, i.e:

```
D:\repos\dotnet_runtime_3\artifacts\obj\System.IO.Pipelines\ref\net6.0-Debug\System.IO.Pipelines.ImplicitNamespaceImports.cs(5,29): error CS0234: The type or namespace name 'Linq' does not exist in the namespace 'System' (are you missing an assembly reference?) [D:\repos\dotnet_runtime_3\src\libraries\System.IO.Pipelines\ref\System.IO.Pipelines.csproj]
```

This was introduced to the SDK last week. https://github.com/dotnet/sdk/commit/85943958e5a01d6bbd39a947aec2a4a37b8ba7cf

cc: @dotnet/runtime-infrastructure

### CODE REVIEW

**Comment by stephentoub** [comment]
I'd convinced myself here:
https://github.com/dotnet/runtime/pull/55828#discussion_r673198075
that libraries were ok.  Your example proves otherwise.  Do you know where my proof breaks down?

**Review by stephentoub — APPROVED** [review]

File: Directory.Build.props [hunk 1]
```diff
@@ -279,6 +279,9 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <!-- By default the SDK produces ref assembly for 5.0 or later -->
     <ProduceReferenceAssembly>false</ProduceReferenceAssembly>
+    <!-- Since we don't use implicit references in most of our projects we need to disable implicit namespace imports
+    to avoid build failures about missing namespaces. -->
+    <DisableImplicitNamespaceImports>true</DisableImplicitNamespaceImports>
```
Comment (stephentoub) [inline Directory.Build.props:284]: Can https://github.com/dotnet/runtime/blob/978b0dbc5e1475be917f4237edf5b47f4c124dba/src/coreclr/tools/Directory.Build.props#L7 be removed?

**Comment by safern** [comment]
> that libraries were ok. Your example proves otherwise. Do you know where my proof breaks down?

Interesting. Let me look at what's going on that it is not working.

**Comment by safern** [comment]
Ah I see. In this case I'm hitting it in `System.IO.Pipelines`, which is an OOB. We only set `DisableImplicitFrameworkReferences` for inbox assemblies, for OOBs we set `DisableImplicitAssemblyReferences`. Which is our own custom thing, so that's why the SDK doesn't set the namespace imports to false:

https://github.com/dotnet/runtime/blob/d38a5396608240a6426668238e06056c69b85c50/eng/targetingpacks.targets#L66-L82

So I believe the right fix is to move this to the location where we set all these settings for libraries and set it to false only for that scenario. 

I think w

File: Directory.Build.props (hunk 1 above)
Comment (safern) [reply Directory.Build.props]: Now that I moved it to where we only need it, we should leave this one as is.

**Review by stephentoub — APPROVED** [review]
Thanks.

**Review by ViktorHofer — COMMENTED** [review]
LGTM 👍

**Pull request merged**